### ISSUE TITLE
Chap 15, pg 513 : ModuleNotFoundError: No module named 'torchdata.datapipes'

### ISSUE THREAD

**Issue by Emmanuel-Ibekwe** [rasbt/machine-learning-book#199]
`
from torchtext.datasets import IMDB
train_dataset = IMDB(split='train')
test_dataset = IMDB(split='test')

`
I keep getting this error despite manually installing torchdata. When I tried installing the exact version of torchtext used in the chapter, version 0.10.0, pip couldn't recognize as a valid version

**Comment by kostuyn** [comment]
@Emmanuel-Ibekwe I installed 0.17.0 version the package and it work (for colab)
`!pip install torchtext==0.17.0 --quiet`

(last version of torchtext has a problem https://github.com/pytorch/text/issues/2272)

**Comment by rasbt** [comment]
@Emmanuel-Ibekwe It looks like you are right, and the PyTorch maintainers removed torchtext 0.10.0 from PyPi for some reason.  The ch15 notebook here on GitHub should be updated to work with newer versions of torchtext though as @kostuyn mentioned. It would require installing `portalocker` as well as described above. Let us know in case this still doesn't work.

**Comment by rasbt** [comment]
Thanks for the feedback. Yes, I think the dataset would nowadays be easier to get from the `datasets` library. The splits are different though, and I am surprised about the low test set accuracy. Both the training and validation accuracy were 100% though? This is an interesting case of overfitting where the validation accuracy seems almost too good to be true (and the test accuracy unexpectedly bad).

**Comment by Emmanuel-Ibekwe** [comment]
Yes, both the training and validation accuracies were 100% at some point during the training process. Here's a link to the code just in case,you want to take a look at the code. [https://github.com/Emmanuel-Ibekwe/Machine-learning-by-S.-Raschka-notebooks/blob/main/Chapter%2015.ipynb](url)
I had commented out the training code and saved the model.

**Comment by rasbt** [comment]
Thanks for sharing, but it seems the link doesn't work: 

<img width="485" alt="Image" src="https://github.com/user-attachments/assets/2d5a04ab-a8ef-4995-af80-c73bf4efc64a" />

**Comment by Emmanuel-Ibekwe** [comment]
Yeah, I noticed that. 
 [https://github.com/Emmanuel-Ibekwe/Machine-learning-by-S.-Raschka-notebooks](https://github.com/Emmanuel-Ibekwe/Machine-learning-by-S.-Raschka-notebooks)

**Comment by rasbt** [comment]
It looks like you are correctly splitting the training set into train and validation subsets. Honestly, I can't see why the accuracies would be exactly the same (100%) in your case. Sorry!

**Comment by Emmanuel-Ibekwe** [comment]
Good day sir. You got something different? You used the hugging face dataset? I've long moved past the chapter though. I got the same accuracies after many iterations.
Thanks for the responses so far.

**Comment by rasbt** [comment]
No worries, I was just looking at your code to see if there was anything suspicious that could explain why the training and validation accuracies would be both 100%. I couldn't find an issue (like an accidentally double-assigned variable or so). In any case, please don't worry about it and feel free to move to the next chapter(s) :).

**Comment by Emmanuel-Ibekwe** [comment]
Ok sir. Thank you for the help so far

**Comment by Emmanuel-Ibekwe** [comment]
Ok sir. I do understand. Thank you for your time.
